experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -160,7 +176,7 @@ export interface SharedTreeList<
 export type SharedTreeObject<
 	TSchema extends StructSchema,
 	API extends "javaScript" | "sharedTree" = "sharedTree",
-> = ObjectFields<TSchema["structFieldsObject"], API>;
+> = ObjectFields<TSchema["structFieldsObject"], API> & SharedTreeNode;
```

Comment:
If the `[nodeSym]` is going to become required, then these types will want to be something like
```suggestion
> = ObjectFields<TSchema["structFieldsObject"], API> & (API extends "sharedTree" ? SharedTreeNode : never);
```
so that users don't need to provide the `[nodeSym]` when making new content. Relatedly, is `[nodeSym]` going to show up in the API? I thought not, in which case - shouldn't these three types avoid explicitly extending `SharedTreeNode`? They will still have the property at runtime, we just won't enforce it in the types.

Reply:
I like the option of not exposing it and see if we encounter any downsides.